Flaky test test_random.test_negative_binomial_generator

test_random.test_negative_binomial_generator failed for unrelated PR #13226 on CI -
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Fwindows-gpu/detail/PR-13226/2/pipeline/

Error log shown below:
```
======================================================================
FAIL: test_random.test_negative_binomial_generator
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Anaconda3\envs\py2\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\jenkins_slave\workspace\ut-python-gpu\tests\python\unittest\common.py", line 173, in test_new
    orig_test(*args, **kwargs)
  File "C:\jenkins_slave\workspace\ut-python-gpu\tests\python\unittest\test_random.py", line 607, in test_negative_binomial_generator
    verify_generator(generator=generator_mx, buckets=buckets, probs=probs)
  File "C:\jenkins_slave\workspace\ut-python-gpu\windows_package\python\mxnet\test_utils.py", line 1966, in verify_generator
    str(buckets), str(probs)))
AssertionError: Generator test fails, Chi-square p=[0.0079457258252656846, 0.015837947946066393, 0.014982295381122585, 0.026195855013954258, 0.016633761926258255], obs_freq=[array([180916, 
241305, 200546, 377233]), array([181918, 242648, 200212, 375222]), array([181759, 241455, 200567, 376219]), array([179932, 243139, 201631, 375298]), array([180884, 241170, 201088, 376858])], 
expected_freq=[array([180800, 242483, 200907, 375809]), array([180800, 242483, 200907, 375809]), array([180800, 242483, 200907, 375809]), array([180800, 242483, 200907, 375809]), 
array([180800, 242483, 200907, 375809])].
buckets=[(-1.0, 2.5), (2.5, 5.5), (5.5, 8.5), (8.5, inf)], probs=[0.18080000000000004, 0.24248320000000032, 0.20090716159999938, 0.37580963840000026]
-------------------- >> begin captured logging << --------------------
common: INFO: Setting test np/mx/python random seeds, use MXNET_TEST_SEED=53079109 to reproduce.
--------------------- >> end captured logging << ---------------------

```

--- a/python/mxnet/test_utils.py
+++ b/python/mxnet/test_utils.py
@@ -1926,7 +1926,7 @@ def chi_square_check(generator, buckets, probs, nsamples=1000000):
     _, p = ss.chisquare(f_obs=obs_freq, f_exp=expected_freq)
     return p, obs_freq, expected_freq
 
-def verify_generator(generator, buckets, probs, nsamples=1000000, nrepeat=5, success_rate=0.25, alpha=0.05):
+def verify_generator(generator, buckets, probs, nsamples=1000000, nrepeat=5, success_rate=0.2, alpha=0.05):
     """Verify whether the generator is correct using chi-square testing.
 
     The test is repeated for "nrepeat" times and we check if the success rate is

--- a/tests/python/unittest/test_random.py
+++ b/tests/python/unittest/test_random.py
@@ -568,7 +568,6 @@ def test_exponential_generator():
                      for _ in range(10)])
             verify_generator(generator=generator_mx_same_seed, buckets=buckets, probs=probs, success_rate=0.20)
 
-@unittest.skip("Flaky test. Tracked in https://github.com/apache/incubator-mxnet/issues/13584")
 @with_seed()
 def test_poisson_generator():
     ctx = mx.context.current_context()

--- a/tests/python/unittest/test_random.py
+++ b/tests/python/unittest/test_random.py
@@ -577,14 +577,13 @@ def test_poisson_generator():
             buckets = [(-1.0, lam - 0.5), (lam - 0.5, 2 * lam + 0.5), (2 * lam + 0.5, np.inf)]
             probs = [ss.poisson.cdf(bucket[1], lam) - ss.poisson.cdf(bucket[0], lam) for bucket in buckets]
             generator_mx = lambda x: mx.nd.random.poisson(lam, shape=x, ctx=ctx, dtype=dtype).asnumpy()
-            verify_generator(generator=generator_mx, buckets=buckets, probs=probs, success_rate=0.2)
+            verify_generator(generator=generator_mx, buckets=buckets, probs=probs)
             generator_mx_same_seed = \
                 lambda x: np.concatenate(
                     [mx.nd.random.poisson(lam, shape=x // 10, ctx=ctx, dtype=dtype).asnumpy()
                      for _ in range(10)])
-            verify_generator(generator=generator_mx_same_seed, buckets=buckets, probs=probs, success_rate=0.2)
+            verify_generator(generator=generator_mx_same_seed, buckets=buckets, probs=probs)
 
-@unittest.skip("Flaky test. Tracked in https://github.com/apache/incubator-mxnet/issues/13506")
 @with_seed()
 def test_negative_binomial_generator():
     ctx = mx.context.current_context()

--- a/tests/python/unittest/test_random.py
+++ b/tests/python/unittest/test_random.py
@@ -577,12 +577,12 @@ def test_poisson_generator():
             buckets = [(-1.0, lam - 0.5), (lam - 0.5, 2 * lam + 0.5), (2 * lam + 0.5, np.inf)]
             probs = [ss.poisson.cdf(bucket[1], lam) - ss.poisson.cdf(bucket[0], lam) for bucket in buckets]
             generator_mx = lambda x: mx.nd.random.poisson(lam, shape=x, ctx=ctx, dtype=dtype).asnumpy()
-            verify_generator(generator=generator_mx, buckets=buckets, probs=probs)
+            verify_generator(generator=generator_mx, buckets=buckets, probs=probs, success_rate=0.2)
             generator_mx_same_seed = \
                 lambda x: np.concatenate(
                     [mx.nd.random.poisson(lam, shape=x // 10, ctx=ctx, dtype=dtype).asnumpy()
                      for _ in range(10)])
-            verify_generator(generator=generator_mx_same_seed, buckets=buckets, probs=probs)
+            verify_generator(generator=generator_mx_same_seed, buckets=buckets, probs=probs, success_rate=0.2)
 
 @unittest.skip("Flaky test. Tracked in https://github.com/apache/incubator-mxnet/issues/13506")
 @with_seed()

